libdokan/archive_reltime_file.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libdokan
+
+import (
+	"time"
+
+	"github.com/keybase/kbfs/libfs"
+	"github.com/keybase/kbfs/libkbfs"
+	"golang.org/x/net/context"
+)
+
+// NewArchiveRelTimeFile returns a special read file that contains a
+// text representation of the global KBFS status.
+func NewArchiveRelTimeFile(
+	fs *FS, handle *libkbfs.TlfHandle, filename string) *SpecialReadFile {
+	return &SpecialReadFile{
+		read: func(ctx context.Context) ([]byte, time.Time, error) {
+			data, isRel, err := libfs.FileDataFromRelativeTimeString(
+				ctx, fs.config, handle, filename)
+			if err != nil {
+				return nil, time.Time{}, err
+			}
+			if !isRel {
+				panic("ArchiveRelTimeFile should only be used with " +
+					"reltime file names")
+			}
+			return data, time.Time{}, nil
```

Comment:
I didn't find an equivalent of `entryValid` for dokan. The closest thing seems to be the [timestamps](https://github.com/keybase/kbfs/blob/537e297f17abe0e7a6d8173574ac008fd05564a9/libdokan/special_read_file.go#L39-L41) in `GetFileInformation`. So perhaps we should return `time.Now()` here?

Reply:
I'm just following the lead of `libfs/status_file.go` and the like, all of which return `time.Time{}`.  I don't think that returned time has anything to do with the cache time of the dir entry (which is what `entryValid` controls for fuse), it's just the atime/ctime/mtime in the fileinfo.